### PR TITLE
Add support for dynamic build matrix generation

### PR DESCRIPTION
**Pull request by Nicoretti**
# ✔ Checklist

* [x] Have you updated the changelog?
* [x] Have you updated the cookiecutter-template?
* [x] Are you mentioning the issue which this PullRequest fixes ("Fixes...")

Note: If any of the above is not relevant to your PR just check the box.

### CODE REVIEW

File: exasol/toolbox/templates/github/workflows/checks.yml [hunk 1]
```diff
@@ -46,14 +46,17 @@ jobs:
...
+
   Lint:
     name: Linting (Python-${{ matrix.python-version }})
     needs: [ Version-Check ]
     runs-on: ubuntu-latest
     strategy:
       fail-fast: false
-      matrix:
-        python-version: [ "3.9", "3.10", "3.11", "3.12" ]
+      matrix: ${{ fromJson(needs.build-matrix.outputs.matrix) }}
```
Comment (tkilias) [inline exasol/toolbox/templates/github/workflows/checks.yml:59]: needs is missing

File: exasol/toolbox/templates/github/workflows/checks.yml [hunk 2]
```diff
@@ -80,8 +83,7 @@ jobs:
...
```
Comment (tkilias) [inline exasol/toolbox/templates/github/workflows/checks.yml:86]: needs is missing

File: exasol/toolbox/templates/github/workflows/checks.yml [hunk 3]
```diff
@@ -160,7 +160,7 @@ jobs:
           python-version: ${{ matrix.python-version }}
 
       - name: Run Tests and Collect Coverage
-        run: poetry run nox -s test:unit -- -- --coverage --db-version ${{ matrix.exasol-version }}
```
[the remark this answers is not in the document]
Comment (Nicoretti) [reply exasol/toolbox/templates/github/workflows/checks.yml:163]: we only do testing over python here. DB relevant tests generally should be in the integration test part imho ...

**Review by Nicoretti — COMMENTED** [review]
-

**Review by tkilias — APPROVED** [review]

**Pull request merged**